core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -303,9 +304,37 @@ static function (array $value): array {
             ->end()
         ;
 
+        $this->addSearchSection($rootNode);
+
         return $treeBuilder;
     }
 
+    private function addSearchSection(ArrayNodeDefinition $rootNode): void
```

Comment:
Why did you encapsulate this in a separate section?

Reply:
Because it‘s best practice (should be done for all the rest as well).

Reply:
That's true, however, you are supposed to use the `append()` method in this case (see https://symfony.com/doc/current/components/config/definition.html#appending-sections).

Reply:
Both ways work just fine🤷‍♂️

Reply:
The current way also works just fine, however, that is not the point. It is a best practice and if you only follow it partly, you might as well not incorporate it at all. 🤓

Reply:
Fixed while merging master into this PR.